src/plugins/discover/public/application/components/top_nav/get_top_nav_links.tsx

```diff
@@ -45,11 +46,9 @@ export const getTopNavLinks = (
       defaultMessage: 'New Search',
     }),
     run() {
-      setTimeout(() => {
-        history().push('/');
-        // TODO: figure out why a history push doesn't update the app state. The page reload is a hack around it
-        window.location.reload();
-      }, 0);
+      core.application.navigateToApp('discover', {
+        path: '#/',
+      });
```

Comment:
Would this reload the page if i was already on discover?

Reply:
Yes it does :)